### PR TITLE
Use mainlinux2010 skip CIBW tests to release packages for GPU

### PR DESCRIPTION
**Pull request by hhorii**
### Summary

Build packages for qiskit-aer-gpu was failed

### Details and comments

mainlinux2014 needs recent GCC versions for qiskit-aer-gpu. Downgrading to mainlinux2010 introduces build of qiskit-terra from source codes, which producing another errors in tests of cibuildwheel.

This PR changes to
- Use mainlinux2010
- Skip tests in cibuildwheel (`CIBW_TEST_SKIP: "cp*"`)

### CODE REVIEW

**Review by mtreinish — COMMENTED** [review]
This shouldn't be necessary because if I remember correctly the devtoolset should already be installed in the manylinux container environment we use for building the wheels. I think the failure is because it's using the wrong base image, I think it should be using manylinux2010 but the failed build tried to use manylinux2014. The issue there is it's installing cuda for centos6 (for manylinux2010), but I haven't had a chance to debug it yet this morning.

File: .github/workflows/build.yml
Comment (mtreinish) [inline .github/workflows/build.yml:176]: FWIW, this won't actually install terra in the right place. Since you're installing it before running cibuildwheel this installs terra in the host environment while cibuildwheel operates in a container which won't see this install. You'd have to do this as part of the the cibuildwheel options to get it to install.

File: .github/workflows/build.yml
Comment (mtreinish) [inline .github/workflows/build.yml:182]: Let's just skip all the tests and move on, the testing doesn't add much here anyway

```suggestion
          CIBW_TEST_SKIP: "cp*"
```

File: .github/workflows/deploy.yml
Comment (mtreinish) [inline .github/workflows/deploy.yml:142]:
```suggestion
          CIBW_TEST_SKIP: "cp*"
```

**Review by mtreinish — COMMENTED** [review]
The issue here is that qiskit-terra only publishes wheels for manylinux 2014 now (for compat with numpy/scipy) and we're building GPU wheels with a manylinux 2010 environment which is too old for those precompiled binaries on pypi. To workaround this we'd need to build terra from source which would require a rust compiler. This shouldn't be hard to do, we use rustup for it in terra's build jobs already so you could just copy that.

But that being said we really can just skip the tests here, the testing on the GPU wheels isn't really all that valuable because we only test the package is installable and that the CPU simulation works (since we have no GPUs available in CI).  We already validate the builds work for the normal `qiskit-aer` wheels so it's very unlikely adding CUDA to the build will break anything so we can just skip tests in CI

**Comment by hhorii** [comment]
@mtreinish Thank you for your review. I think we should not include `wheel-gpu` test in test.yml because it takes mostly one hour to build gpu packages. This PR only includes change of deploy.yml to switch base linux image and add CIBW_TEST_SKIP.

**Review by mtreinish — APPROVED** [review]
LGTM, now after this merges I'll push up a branch with this to manually trigger a GPU wheel build to get a 0.11.0 version published

**Pull request merged**
